frontend/src/js/external-forms/helper.ts

```diff
@@ -1,10 +1,27 @@
 import type { SelectOptionT } from "../api/types";
 import type { Language } from "../localization/useActiveLang";
 
-import type { FormField, GeneralField, Group } from "./config-types";
+import type {
+  Field,
+  FormField,
+  GeneralField,
+  Group,
+  Tabs,
+} from "./config-types";
 
 const nonFormFieldTypes = new Set(["HEADLINE", "DESCRIPTION"]);
 
+// Different forms may have fields with the same name.
+// We want to remember values of fields of form A when switching to form B,
+// so users may come back to form A and see their previous values.
+// So in order to avoid field name clashes, we need unique field names
+export const getUniqueFieldname = (formType: string, field: Field | Tabs) => {
+  return `${formType}--${field.name}`;
```

Comment:
There might be an even better place for this (where you fetch the field names), where you assert that no field contains `--`  since that would break getRawFieldName.

Reply:
Hmm, yeah I thought about transforming directly after the fetch, not to prevent some accidental `--` within field names (I think it's a fair assumption / contract that they won't), but it would indeed avoid making the transformation at many places in the code. I'll give it a try